#! /usr/bin/env python
# coding=utf-8

import roslib
from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE = "pure_pursuit_fp"
roslib.load_manifest(PACKAGE)

# 创建一个参数生成器
gen = ParameterGenerator()

# 添加参数说明，便于后续生成界面
#      参数名        类型           等级       参数描述                默认值          最小值         最大值
gen.add("kp",        double_t,        0,        "KP param.",            1.0,            -100.00,        100.00)
gen.add("ki",        double_t,        0,        "KI param.",            0.01,            -100.00,        100.00)
gen.add("kd",        double_t,        0,        "KD param.",            0.005,            -100.00,        100.00)
gen.add("max_speed", double_t,        0,        "Max speed.",           1.3,            -100.00,        100.00)
gen.add("min_ld",    double_t,        0,        "Min look ahead dis.",  0.8,            -100.00,        100.00)
gen.add("car_wheel_base",double_t,        0,        "Car wheel base.",      0.2,            -100.00,        100.00)
gen.add("control_rate",int_t,      0,        "Controller rate.",     50,            -100,            100)

# 调用生成器生成config配置文件
#                 包名      节点名称    生成文件名
exit(gen.generate(PACKAGE, PACKAGE, "PID"))